tests/QueryBuilder/DSL/AbstractDSLTest.php

```diff
@@ -67,21 +67,16 @@ protected function _getDefaultValue(\ReflectionParameter $param)
         return null;
     }
 
-    /**
-     * @return string|null
-     */
-    protected function _getHintName(\ReflectionParameter $param)
+    protected function _getHintName(\ReflectionParameter $param): ?string
     {
-        if ($param->isCallable()) {
-            return 'callable';
-        }
-
-        if ($param->isArray()) {
-            return 'array';
+        if (null === $type = $param->getType()) {
```

Comment:
I must confess I'm still trying to get my head around this change. It does not change anything from a user perspective?

If this line would be removed, would it also be catched by line 82? Or would in_array blow up it type is null?

Reply:
Sorry for the headache 😉
This line is just to ensure that `$type` is not null, so that it can be used safely below.

Reply:
Parameter type is stored inside `$type` to avoid calling the getter multiple time